randovania/games/super_metroid/patcher/super_duper_metroid_patcher.py

```diff
@@ -54,24 +57,125 @@ def sm_pickup_details_to_patcher(detail: pickup_exporter.ExportedPickupDetails,
+    _mapping = {
+        "Missile": "Missile Expansion",
+        "Super Missile": "Super Missile Expansion",
+        "Power Bombs": "Power Bomb Expansion",
+
+        "Nothing": "No Item",
     }
 
+    _effect = {
+        "Energy Tank": "Get Energy Tank",
+        "Missile Expansion": "Get Missile Expansion",
+        "Super Missile Expansion": "Get Super Missile Expansion",
+        "Power Bomb Expansion": "Get Power Bomb Expansion",
+        "Grapple Beam": "Get Grapple Beam",
+        "X-Ray Scope": "Get X-Ray Scope",
+        "Varia Suit": "Get Varia Suit",
+        "Spring Ball": "Get Spring Ball",
+        "Morph Ball": "Get Morph Ball",
+        "Screw Attack": "Get Screw Attack",
+        "Hi-Jump Boots": "Get Hi-Jump Boots",
+        "Space Jump": "Get Space Jump",
+        "Speed Booster": "Get Speed Booster",
+        "Charge Beam": "Get Charge Beam",
+        "Ice Beam": "Get Ice Beam",
+        "Wave Beam": "Get Wave Beam",
+        "Spazer Beam": "Get Spazer Beam",
+        "Plasma Beam": "Get Plasma Beam",
+        "Morph Ball Bombs": "Get Morph Ball Bombs",
+        "Reserve Tank": "Get Reserve Tank",
+        "Gravity Suit": "Get Gravity Suit",
+        "No Item": "No Effect",
+    }
+
+    item_name = _mapping.get(item.long_name, item.long_name)
+
+    result = {
+        "item_name": item_name,
+        "quantity_given": quantity,
+        "pickup_effect": _effect[item_name],
+    }
     return result
 
 
-def _json_to_pickup_placement_data(pickup: dict) -> SuperDuperMetroid.ROM_Patcher.PickupPlacementData:
-    return SuperDuperMetroid.ROM_Patcher.PickupPlacementData(
-        quantity_given=pickup["quantity"],
-        pickup_index=pickup["pickup_index"],
-        item_name=pickup["item_name"],
-    )
+def sm_starting_save_station_to_patcher(location: AreaIdentifier) -> dict:
+    _starting_area_data = {
+        "Brinstar: Big Pink Save Station": 0,
+        "Brinstar: Green Brinstar Upper Save Station": 1,
+        "Brinstar: Green Brinstar Lower Save Station": 2,
+        "Brinstar: Warehouse Save Station": 3,
+        "Brinstar: Red Tower Save Station": 4,
+        "Ceres Station: Elevator Shaft": 0,
+        "Crateria: Landing Site": 0,
+        "Crateria: Parlor Save Station": 1,
+        "Maridia: Maridia Glass Tube Save Station": 0,
+        "Maridia: Maridia Elevator Save Station": 1,
+        "Maridia: Aqueduct Save Station": 2,
+        "Maridia: Colosseum Save Station": 3,
+        "Norfair: Post Crocomire Save Station": 0,
+        "Norfair: Bubble Mountain Save Station": 1,
+        "Norfair: Business Center Save Station": 2,
+        "Norfair: Crocomire Save Station": 3,
+        "Norfair: Elevator to Lower Norfair Save Station": 4,
+        "Norfair: Red Kihunter Shaft Save Station": 5,
+        "Tourian: Rinka Shaft Save Station": 0,
+        "Tourian: Upper Tourian Save Station": 1,
+        "Wrecked Ship: Shaft Save Station": 0,
+    }
```

Comment:
I recommend including this data in the `extra` field for each of these areas in the database instead of defining it in a list like this